File: Leetcode/C/reverse_linked_list.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

struct ListNode {
	int val;
	struct ListNode* next;
};





struct ListNode* reverseListX(struct ListNode* head) {
	struct ListNode* reversed = head, * temp;

	while (reversed && reversed->next) {
		temp = reversed->next;
		if (reversed->next->next)
			reversed->next = reversed->next->next;
		else
			reversed->next = NULL;
		temp->next = head;
		head = temp;
	}
	return head;
}

File: Leetcode/C#/LinkedList/reverse_linked_list.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LeetCode.LinkedList
{
    class reverse_linked_list
    {
		public ListNode ReverseList(ListNode head)
		{
			ListNode reversed = head, temp;

			while (reversed != null && reversed.next != null)
            {
				temp = reversed.next;

				if (reversed.next.next != null)
					reversed.next = reversed.next.next;
				else
					reversed.next = null;


				temp.next = head;
				head = temp;
            }

			return head;
		}
	}
}
